lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -169,11 +169,7 @@ def merge_json_log(record)
         log = record['log'].strip
         if log[0].eql?('{') && log[-1].eql?('}')
           begin
-            parsed_log = JSON.parse(log)
-            record     = record.merge(parsed_log)
-            unless parsed_log.has_key?('log')
-              record.delete('log')
-            end
+            record = JSON.parse(log).merge(record)
```

Comment:
This flips the merge logic effectively from `record.merge(log)` to `log.merge(record)`. Which way round do we want to do this? Can't remember what the semantics for this are - only matters for conflicting keys I guess.